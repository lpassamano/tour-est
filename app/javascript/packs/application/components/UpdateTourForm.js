import React, { Component } from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import TourFormFields from "./TourFormFields";
import * as tourActions from "../redux/tours/actions";
import * as tourSelectors from "../redux/tours/selectors";

export class UpdateTourForm extends Component {
  handleSubmit = attributes => {
    this.props.onUpdateTour({ tour: attributes });
  };

  render() {
    const { isFetching, tour } = this.props;
    if (isFetching || !tour) {
      return <p>loading... please wait!</p>;
    }

    return (
      <TourFormFields
        onSubmit={this.handleSubmit}
        initialValues={this.props.tour}
      />
    );
  }
}

UpdateTourForm.propTypes = {
  onUpdateTour: PropTypes.func.isRequired,
  tourId: PropTypes.string.isRequired,
  tour: PropTypes.shape({
    id: PropTypes.number.isRequired,
    title: PropTypes.string.isRequired,
    starting_point: PropTypes.string,
    directions: PropTypes.string,
    estimated_time: PropTypes.string,
    description: PropTypes.string
  })
};

const mapStateToProps = (state, ownProps) => ({
  tour: tourSelectors.getTour(state, ownProps.tourId),
  isFetching: tourSelectors.isFetching(state)
});

const mapDispatchToProps = { onUpdateTour: tourActions.updateTour };

const enhance = connect(
  mapStateToProps,
  mapDispatchToProps
);

export default enhance(UpdateTourForm);
