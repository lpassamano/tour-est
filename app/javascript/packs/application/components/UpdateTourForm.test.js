import { UpdateTourForm } from "./UpdateTourForm";
import React from "react";
import { shallow } from "enzyme";

describe("<UpdateTour Form/>", () => {
  const setup = () => {
    const tour = { title: "Tour 1", id: 1 };
    const onUpdateTour = jest.fn();
    const isFetching = false;
    const component = shallow(
      <UpdateTourForm
        isFetching={isFetching}
        onUpdateTour={onUpdateTour}
        tour={tour}
        tourId={"1"}
      />
    );
    return component;
  };

  test("render", () => {
    const component = setup();
    expect(component).toMatchSnapshot();
  });
});
