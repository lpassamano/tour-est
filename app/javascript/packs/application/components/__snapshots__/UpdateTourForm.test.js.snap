// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UpdateTour Form/> render 1`] = `
<TourFormFields
  initialValues={
    Object {
      "id": 1,
      "title": "Tour 1",
    }
  }
  onSubmit={[Function]}
/>
`;
